### DISCUSSION TITLE
Perform code before render dialog with custom component

### DISCUSSION THREAD

**Discussion by cirolosapio** [quasarframework/quasar#10547] [General - Components / Directives / etc]
With the Options API when we use a dialog with custom component we can perform code before render the dialog:
```s
show () {
  // code
  this.$refs.dialog.show()
}
``` 

Is it possible to do this also with the Composition API?

maybe exposing the show / hide methods

**Comment by metalsadman** [comment]
check here in the docs, there are variants example further down for composition and options api. https://quasar.dev/quasar-plugins/dialog#invoking-custom-component

**Reply by cirolosapio** [reply]
> 
> 
> check here in the docs, there are variants example further down for composition and options api. https://quasar.dev/quasar-plugins/dialog#invoking-custom-component

yes, I have seen the examples, but there is no way to run code before opening the dialog with the Composition API

**Reply by metalsadman** [reply]
What have you tried so far? Post in a codepen if possible.
